### ISSUE TITLE
Please create upstart service for cerebro

### ISSUE THREAD

**Issue by ahazelwood** [lmenezes/cerebro#87]
I would love to be able to have cerebro run startup.  However I can't figure out how to create a service template for cerebro.

Thanks.

**Comment by IgorZet** [comment]
I've created /etc/init.d/ script for cerebro. May share it here

**Comment by Amorik** [comment]
For systemd servers (CentOS 7) you can create service script relatively easily.

vi /etc/systemd/system/cerebro.service

`[Unit]
Description=Cerebro

[Service]
Type=simple
User=root
Group=root
ExecStart=/<install dir>/bin/cerebro "-Dpidfile.path=/var/run/cerebro.pid"
Restart=always
WorkingDirectory=/<install dir>

[Install]
WantedBy=multi-user.target`


systemctl daemon-reload
systemctl enable cerebro
systemctl start cerebro

~Regards

**Comment by yanyov** [comment]
@ahazelwood 
For upstart you can use:

# Elasticsearch cerebro plugin

description     "Cerebro plugin"

start on runlevel [2345]
stop on runlevel [06]

script
  exec start-stop-daemon --start --quiet --exec /usr/local/cerebro-0.6.4/bin/cerebro
end script

Just change the path to the bin.

**Comment by jptoto** [comment]
@Amorik the cerebro.service should be in /usr/lib/systemd/service for Centos 7.x, no?

**Comment by Amorik** [comment]
@jptoto probably? both work for me, but I'm sure one is more "right" than the other.

**Comment by hameno** [comment]
My upstart script:
```
#!upstart
description "Cerebro"

start on startup
stop on shutdown

exec start-stop-daemon --start --make-pidfile --pidfile /var/run/cerebro.pid --chuid cerebro --exec /opt/cerebro/bin/cerebro -- -Dhttp.address=127.0.0.1 >> /var/log/cerebro.log 2>&1
```

**Comment by alkuzad** [comment]
@Amorik I think root is not necessary for cerebro, I strongly recommend to use own user for it's purposes. Also why you set WorkingDirectory to / ?

**Comment by Amorik** [comment]
@alkuzad Agreed. It was a hasty copy from one of my dev boxes. As for the working directory no reason. In prod it was placed nearby the install folder.

**Comment by frenkye** [comment]
Hi, if anyone interested I reworked init script for kibana to work for cerebro. Nothing special you just need create user to run cerebro and of course change rights on directory.


[cerebro.init.txt](https://github.com/lmenezes/cerebro/files/1182579/cerebro.init.txt)

**Comment by druzmieres** [comment]
@IgorZet could you please share your script? I'm trying to do the same and would like to see it.

**Comment by peelman** [comment]
kudos to @Amorik for his `systemd` script.  It works wonderfully on Ubuntu as well.

For reference, my Cerebro lives in `/opt`, and is symlinked to `/opt/current`, so my version looks like this:

```
npeelman@oliver:~$ cat /etc/systemd/system/cerebro.service 
[Unit]
Description=Cerebro

[Service]
Type=simple
User=root
Group=root
ExecStart=/opt/cerebro/current/bin/cerebro "-Dpidfile.path=/var/run/cerebro.pid"
Restart=always
WorkingDirectory=/opt/cerebro

[Install]
WantedBy=multi-user.target
npeelman@oliver:~$ 
```

**Comment by wols** [comment]
```
# useradd -r -b /opt -m cerebro
# ll /opt/
drwx------. 2 cerebro cerebro 59 26. Oct 19:40 cerebro
# id cerebro
uid=993(cerebro) gid=990(cerebro) groups=990(cerebro)
```
```
...
User=cerebro
Group=cerebro
...
```
